core/src/main/java/io/dekorate/kubernetes/decorator/AddDockerConfigJsonSecretDecorator.java

```diff
@@ -67,7 +67,7 @@ public AddDockerConfigJsonSecretDecorator(String name, Path existingConfigJson,
 
   @Override
   public void visit(KubernetesListBuilder list) {
-    String name = Strings.isNotNullOrEmpty(this.name) ? this.name : getMandatoryDeploymentMetadata(list).getName();
+    String name = Strings.isNotNullOrEmpty(this.name) ? this.name : getMandatoryDeploymentMetadata(list, ANY).getName();
```

Comment:
I would read this line easier with `getDeploymentMetadata(list, ANY).map(ObjectMeta::getName).orElse(name)`